rfcs/0054-support-levels-and-tiers.md

```diff
@@ -0,0 +1,172 @@
+---
+feature: support_levels_and_tiers
+start-date: 2019-10-03
+author: CrystalGamma
+co-authors: –
+related-issues: –
+---
+
+# Summary
+[summary]: #summary
+
+A scheme for expressing and managing support of packages and for platforms in nixpkgs.
+
+# Motivation
+[motivation]: #motivation
+
+Currently there is no consensus in the nixpkgs maintainer community regarding how to deal with platform-specific code and changes, as well as few official statements and processes to represent platform support to users.
+As a growing distribution, the Nix community needs to establish a framework to determine platform support and decide what platform-specific packaging code is acceptable.
+
+Providing a good base for platform discussions and defining support expectations should improve communication with users (especially those of non-mainstream platforms) and should allow establishing procedures to deal with developers wishing to add support for new architectures.
+
+# Detailed design
+[design]: #detailed-design
+
+This RFC defines scopes for packages and the corresponding support and acceptance tiers for platforms.
+
+## Support Level
+The *scope* of a package is a designator of its importance for nixpkgs.
+
+This RFC proposes an initial set of 4 scopes (in order of descending importance): `core`, `bootstrap`, `system` and `all`.
+(Terminology: A scope to the left of another in this list is said to be 'narrower' or 'more significant' than it. Conversely a scope to the right of another is considered 'wider'/'broader' or 'less significant'.) Each broader scope includes the packages of the narrower ones. See [below](#relationship-between-scope-support-and-acceptance) for what by each scope comprises.
+
+A `meta` attribute `scope` which is supposed to take a value from the aforementioned enumeration is introduced for this purpose.
+A package with its `meta.scope` unset is assumed to have scope `all` (in fact, setting `scope` to `all` should only be done in special cases, such as for packages produced by overriding others).
+
+Scrutiny required for packages increases with narrower scope, as such the scope for each package should be kept as broad as possible.
+
+## Platforms and support tiers
+A platform is a combination of these differentiations:
+
+* CPU architecture
+* kernel
```

Comment:
This is a minor nitpick, but I think the phrase "OS ABI" is more appropriate than kernel. For example, FreeBSD and Fuchsia can emulate the Linux ABI, CloudABI binaries are basically kernel-agnostic.